packages/sdk-core/src/sourcemaps/DebugIdProvider.ts

```diff
@@ -0,0 +1,45 @@
+import { BacktraceStackTraceConverter } from '../modules/converter';
+import { DebugIdMapProvider } from './interfaces/DebugIdMapProvider';
+
+export const SOURCE_DEBUG_ID_VARIABLE = '_btDebugIds';
+
+export class DebugIdProvider {
+    private _fileDebugIds?: Record<string, string>;
+
+    constructor(
+        private readonly _stackTraceConverter: BacktraceStackTraceConverter,
+        private readonly _debugIdMapProvider?: DebugIdMapProvider,
+    ) {}
```

Comment:
why we cannot file fileDebugIds in the constructor? Initialization time is a great moment to do that, instead of doing it on demand when the record happened. 


Why?
We have limited time/resources when an exception occurs. In worst case scenario it might take a while to load it and later reprocess.

Reply:
Changed.